src/main/kotlin/app/ryss/service/metrics/MemoryMetrics.kt

```diff
@@ -0,0 +1,56 @@
+package app.ryss.service.metrics
+
+import com.influxdb.client.InfluxDBClient
+import com.influxdb.client.domain.WritePrecision
+import com.influxdb.client.write.Point
+import kotlinx.coroutines.ObsoleteCoroutinesApi
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.channels.ticker
+import mu.KotlinLogging
+import java.net.InetAddress
+import java.time.Instant
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+/**
+ * MemoryMetrics posts the amount of total heap and the amount of allocated heap every second to InfluxDB.
+ * @param client the [InfluxDBClient]
+ * @param bucket the InfluxDB bucket to post stats to
+ * @param org the InfluxDB org to post stats to
+ */
+class MemoryMetrics(private val client: InfluxDBClient, private val bucket: String, private val org: String) {
+
+    private val log = KotlinLogging.logger { }
+
+    private val context = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
+
+    @OptIn(ObsoleteCoroutinesApi::class)
```

Comment:
I think you need to add `Xopt-in=kotlin.RequiresOptIn` as a compiler argument to the build.gradle.kts